lib/shopify_api/rest/request.ex

```diff
@@ -72,15 +75,43 @@ defmodule ShopifyAPI.REST.Request do
 
   ## Private Helpers
 
-  defp log_request(method, url, time, response) do
+  defp send_telemetry(
+         %{app_name: app, shop_name: shop},
+         method,
+         url,
+         time,
+         {:ok, %{status_code: status}} = response
+       ) do
+    :telemetry.execute(
+      [:shopify_api, :rest, :request],
+      %{request_time: time, call_limit: call_limit(response)},
```

Comment:
I'm not certain `call_limit` is an accurate name for this measurement, maybe `remaining_calls` is better?